client/src/main/scala/org/http4s/client/ConnectionFailure.scala

```diff
@@ -1,11 +1,21 @@
 package org.http4s.client
 
 import java.io.IOException
+import java.net.InetSocketAddress
 
 /** Indicates a failure to establish a client connection, preserving the request key
   * that we tried to connect to.
   */
-class ConnectionFailure(requestKey: RequestKey, cause: Throwable) extends IOException(cause) {
+class ConnectionFailure(
+    val requestKey: RequestKey,
+    val upstream: InetSocketAddress,
```

Comment:
@ChristopherDavenport questioned on Gitter whether we'd always have this in other backends where we might wish to wrap exceptions.  We might want to make this an `Option` for portability, though I haven't tried to thread this into other backends yet.

Reply:
up to you, im ok with it being optional